Modules/_hashopenssl.c

```diff
@@ -355,7 +355,7 @@ py_digest_by_name(PyObject *module, const char *name, enum Py_hash_type py_ht)
         }
     }
     if (digest == NULL) {
-        _setException(PyExc_ValueError, "unsupported hash type %s", name);
+        _setException(state->unsupported_digestmod_error, "unsupported hash type %s", name);
```

Comment:
nice.  looking around, as this is already a ValueError subclass, this is indeed the ideal fix for this problem. :)